### PR TITLE
feat: DBTP-1568 - Add s3 support for external role access

### PR DESCRIPTION
**Pull request by ksugden**
Addresses [DBTP-1568](https://uktrade.atlassian.net/browse/DBTP-<ticket>) adding validation and schema for s3 external role access configuration block.  Feature is documented here:
https://github.com/uktrade/platform-documentation/pull/422
---
### Title:
- [x] Scope included as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Ticket reference included (unless it's a quick out of ticket thing)
### Description:
- [x] Link to ticket included (unless it's a quick out of ticket thing)
- [x] Includes tests (or an explanation for why it doesn't)
- [ ] If the work includes user interface changes, before and after screenshots included in description
- [ ] Includes any applicable changes to the documentation in this code base
- [x] Includes link(s) to any applicable changes to the documentation in the [DBT Platform Documentation](https://platform.readme.trade.gov.uk/) (can be to a pull request)
### Tasks:
- [x] [Run the end to end tests for this branch]([https://github.com/uktrade/platform-tools?tab=readme-ov-file#regression-tests](https://github.com/uktrade/platform-end-to-end-tests?tab=readme-ov-file#running-the-tests)) and confirm that they are passing

[DBTP-1568]: https://uktrade.atlassian.net/browse/DBTP-1568?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by JohnStainsby — APPROVED** [review]

File: dbt_platform_helper/utils/validation.py [hunk 1]
```diff
@@ -266,6 +266,25 @@ def iam_role_arn_regex(key):
     )
 
 
+def dbt_email_address_regex(key):
+    return Regex(
+        r"^[\w\-\.]+@(businessandtrade.gov.uk|digital.trade.gov.uk)$",
```
Comment (antroy-madetech) [inline dbt_platform_helper/utils/validation.py:271]: I don't think you need to escape the . and hyphen in the character group, but the hyphen will probably need to come last to stop the regex parser from thinking it's a character range.
```suggestion
        r"^[\w.-]+@(businessandtrade.gov.uk|digital.trade.gov.uk)$",
```

Are we sure that these are the only characters used in DBT email addresses?

File: dbt_platform_helper/utils/validation.py [hunk 2]
```diff
@@ -266,6 +266,25 @@ def iam_role_arn_regex(key):
...
+
+EXTERNAL_ROLE_ACCESS = {
+    "role_arn": iam_role_arn_regex("role_arn"),
+    "read": bool,
+    "write": bool,
+    "cyber_sign_off_by": dbt_email_address_regex("cyber_sign_off_by"),
+}
+
+EXTERNAL_ROLE_ACCESS_NAME = Regex(
+    r"^([a-z][a-zA-Z0-9_-]*|\*)$",
```
Comment (antroy-madetech) [inline dbt_platform_helper/utils/validation.py:284]: This allows for the external role access name to be just `*`?

File: dbt_platform_helper/utils/validation.py [hunk 3]
```diff
@@ -266,6 +266,25 @@ def iam_role_arn_regex(key):
...
+    error="External role access block name {} is invalid: names must only contain lowercase alphanumeric characters separated by hypen or underscore",
```
Comment (antroy-madetech) [inline dbt_platform_helper/utils/validation.py:285]: In which case we should allow for that in the error message.

**Review by antroy-madetech — COMMENTED** [review]
Initial comments - will probably fix some of these as I start on the next stuff.

File: dbt_platform_helper/utils/validation.py (hunk 1 above)
Comment (ejayesh) [reply dbt_platform_helper/utils/validation.py:271]: yes, we only use these two.  Eveyones old address have been moved over to businessandtrade

File: dbt_platform_helper/utils/validation.py (hunk 2 above)
Comment (ksugden) [reply dbt_platform_helper/utils/validation.py]: that was a copy-paste error from ENV regex.  I've removed the * option

**Pull request merged**